wolfcrypt/src/wc_pkcs11.c

```diff
@@ -415,19 +415,29 @@ int wc_Pkcs11_Initialize(Pkcs11Dev* dev, const char* library, void* heap)
     if (ret == 0) {
         dev->func = NULL;
         func = dlsym(dev->dlHandle, "C_GetFunctionList");
-        if (func == NULL)
+        if (func == NULL) {
+            WOLFSSL_MSG(dlerror());
             ret = WC_HW_E;
+        }
     }
     if (ret == 0) {
-        if (((CK_C_GetFunctionList)func)(&dev->func) != CKR_OK)
+        if ((ret = ((CK_C_GetFunctionList)func)(&dev->func)) != CKR_OK) {
```

Comment:
Please split the call and the if into separate statements.
And below.

Reply:
Thanks, gotcha.